src/Loading/Plugins/babylon.babylonFileLoader.ts

```diff
@@ -269,23 +277,25 @@
                 }
                 
                 //collisions, if defined. otherwise, default is true
-                if (parsedData.collisionsEnabled != undefined) {
+                if (parsedData.collisionsEnabled !== undefined) {
                     scene.collisionsEnabled = parsedData.collisionsEnabled;
                 }
                 scene.workerCollisions = !!parsedData.workerCollisions;
 
                 var index: number;
                 var cache: number;
                 // Lights
-                for (index = 0, cache = parsedData.lights.length; index < cache; index++) {
-                    var parsedLight = parsedData.lights[index];
-                    var light = Light.Parse(parsedLight, scene);
-                    log += (index === 0 ? "\n\tLights:" : "");
-                    log += "\n\t\t" + light.toString(fullDetails);
+                if (parsedData.lights !== undefined) {
+                    for (index = 0, cache = parsedData.lights.length; index < cache; index++) {
+                        var parsedLight = parsedData.lights[index];
+                        var light = Light.Parse(parsedLight, scene);
+                        log += (index === 0 ? "\n\tLights:" : "");
+                        log += "\n\t\t" + light.toString(fullDetails);
+                    }
                 }
     
                 // Animations
-                if (parsedData.animations) {
+                if (parsedData.animations !== undefined) {

```

Comment:
I'd be happy to make that fix. Does that logic apply to all property checks or is the 'animation' property unique?

My primary motivation to all checks being a strict undefined check is that there seems there could be time for some of the other properties being 1, true or similar value would result if a false positive. My secondary motivations are code clarity, slight JS performance and consistency between the checks in this block of code.

Alternatively we could keep it strict and cover both null and undefined, but not leave it open to false positives: `if (parsedData.animations !== undefined && parsedData.animations !== null) {`

This is your project, and I am not familiar enough with the code base so I rely on and appreciate your guidance. Change just "animation" back to the more forgiving condition, should I change other properties back, or add the null check to all cases?

Thanks!